src/CPUSolver.cpp

```diff
@@ -907,8 +1025,51 @@ void CPUSolver::transferAllInterfaceFluxes() {
     _timer->stopTimer();
     _timer->recordSplit("Packing time");
 
-    /* Send and receive from all neighboring domains */
+#ifdef ONLYVACUUMBC
+    /* Check if any rank needs to send buffers : because of the pre-filling
+       some nodes might have sent all their fluxes before others */
+    int need_to_send = 0;
+    for (int i=0; i < num_domains; i++) {
+
+      long* first_track_idx =
+        reinterpret_cast<long*>(&_send_buffers.at(i)[_fluxes_per_track+1]);
+      long first_track = first_track_idx[0];
+      if (first_track != -1)
+        need_to_send = 1;
```

Comment:
A break of the for loop could be put here avoiding extra computation.

Reply:
There's not much work done in that loop and not many iterations, it's fine to keep going